mantidproject/mantid
# SumSpectra: Fixed one-off errors & input validation

Fixes #15414.

Fixed the one-off errors in SumSpectra's min/max fallback indices as described in the ticket. Also added input validation for the max index and index list properties which should prevent the fallback values from being needed in the first place.

**To test:**
1. Create sample workspace
1. Run SumSpectra on it 
1. Try entering invalid values (i.e. negative, or greater than maximum spectrum index) for StartWorkspaceIndex, EndWorkspaceIndex or ListOfWorkspaceIndices properties
1. Confirm an error appears when trying to run the algorithm, and the tooltip for the invalid input is understandable
1. Confirm functionality of the algorithm is unchanged for valid inputs


[Release Notes](https://github.com/mantidproject/mantid/pull/19830/commits/ad572aa8b8b353f19fcdead786f8fe763f69b468)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
